docs/source/developer_guide.rst

```diff
@@ -0,0 +1,67 @@
+===============
+Developer guide
+===============
+
+This guide provides an overview of the software design of **pansat** and outlines
+the responsibilities of different parts of the code.
+
+Code structure
+==============
+
+The **pansat** package currently has 4 submodules with different responsibilities:
+
+  1. The :py:mod:`pansat.products` contains the product class representing different
+     data products. These classes provide the main interface for the user to
+     download data.
+  2. The :py:mod:`pansat.download` module contains the code which is responsible for
+     downloading the data for different products. It acts as an abstraction layer
+     to decouple the data products from their download location.
+  3. The :py:mod:`pansat.formats` module contains interfaces to open data from
+     different file formats.
+  4. The :py:mod:`pansat.catalogue` module provides functionality to catalogue downloaded
+     satellite data.
+
+For more information on each submodule, refer to the respective source code documentation.
+
+
+Key design concepts
+===================
+
+**pansat** aims to provide a unified interface to download and access a wide
+range of satellite, reanalysis and other meteorological data. Unfortunately,
+this data and how to access it varies widely making and is difficult to handle
+in a consistent way. In an attempt to overcome these difficulties, **pansat** uses
+an object-oriented design to decouple its components and make it sufficiently
+expressive and flexible to handle a wide range of data products.
+
+Products and data providers
```

Comment:
Not sure whether this is maybe redundant, but in this section, it could be clarified that the three steps of adding a new data product for download when the provider does not exist is 
1. add new module with provider class in pansat.download.providers
2. add identity to accounts.py with add_identity (do developers here need to know more about how they could create a pull request with a suggestion for new providers and products without uploading their personal account info?)
3. add new module with product class in pansat.products